mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -3734,6 +3805,198 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       };
       auto weiGemmGGemmKGemmM = getWeiGemmGGemmKGemmM(secondFilterDimName);
 
+      // below is padding kernel for backward data
```

Comment:
could we refactor the code so they are in a separate function, or even a separate header file, just like what @ltqin  did in #271 ?

this could help us identify reusable modules which could help our next round of refactoring.